cmd/checkpoint/main.go

```diff
@@ -0,0 +1,268 @@
+				}
+			}
+		}
+		time.Sleep(60 * time.Second)
+	}
+}
+
+func getPodsFromKubeletAPI() []byte {
+	var pods []byte
+	res, err := http.Get(kubeletAPIPodsURL)
+	if err != nil {
+		log.Println(err)
+		return pods
+	}
+	pods, err = ioutil.ReadAll(res.Body)
+	res.Body.Close()
+	if err != nil {
+		log.Println(err)
+	}
+	return pods
+}
+
+func bothAPIServersRunning(pods v1.PodList) bool {
+	var kubeAPISeen, tempAPISeen bool
+	for _, p := range pods.Items {
+		switch {
+		case isKubeAPI(p):
+			kubeAPISeen = true
+		case isTempAPI(p):
+			tempAPISeen = true
+		}
+		if kubeAPISeen && tempAPISeen {
+			return true
+		}
+	}
+	return false
+}
+
+func kubeSystemAPIServerRunning(pods v1.PodList) bool {
+	for _, p := range pods.Items {
+		if isKubeAPI(p) {
+			return true
+		}
+	}
+	return false
+}
+
+func isKubeAPI(pod v1.Pod) bool {
+	return strings.Contains(pod.Name, "kube-apiserver") && pod.Namespace == api.NamespaceSystem
+}
+
+func isTempAPI(pod v1.Pod) bool {
+	return strings.Contains(pod.Name, "temp-apiserver") && pod.Namespace == api.NamespaceSystem
+}
+
+// cleanVolumes will sanitize the list of volumes and volume mounts
+// to remove the default service account token.
+func cleanVolumes(p *v1.Pod) {
+	volumes := make([]v1.Volume, 0, len(p.Spec.Volumes))
+	for _, v := range p.Spec.Volumes {
+		if !strings.Contains(v.Name, "default") {
+			volumes = append(volumes, v)
+		}
+	}
+	p.Spec.Volumes = volumes
+	for i := range p.Spec.Containers {
+		c := &p.Spec.Containers[i]
+		volumeMounts := make([]v1.VolumeMount, 0, len(c.VolumeMounts))
+		for _, vm := range c.VolumeMounts {
+			if !strings.Contains(vm.Name, "default") {
+				volumeMounts = append(volumeMounts, vm)
+			}
+		}
+		c.VolumeMounts = volumeMounts
+	}
+}
+
+func modifyInsecurePort(p *v1.Pod) {
+	for i := range p.Spec.Containers {
+		c := &p.Spec.Containers[i]
+		cmds := c.Command
+		for i, cmd := range cmds {
+			if strings.Contains(cmd, "insecure-port") {
+				cmds[i] = strings.Replace(cmd, "8080", "8081", 1)
+				break
+			}
+		}
+	}
+}
+
+// writeManifest will write the manifest to the ignore path.
+// It first writes the file to a temp file, and then atomically moves it into
+// the actual ignore path and correct file name.
+func writeManifest(manifest v1.Pod) {
+	m, err := json.Marshal(manifest)
+	if err != nil {
+		log.Fatal(err)
+	}
+	writeAndAtomicCopy(m, ignoreFile)
+}
+
+func parseAPIPodSpec(podList v1.PodList) v1.PodSpec {
+	var apiPod v1.Pod
+	for _, p := range podList.Items {
+		if strings.Contains(p.Name, string(kubeAPIServer)) {
+			apiPod = p
+			break
+		}
+	}
+	cleanVolumes(&apiPod)
+	modifyInsecurePort(&apiPod)
+	return apiPod.Spec
+}
+
+func newAPIClient() clientset.Interface {
+	kubeConfig, err := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(&clientcmd.ClientConfigLoadingRules{
+		ExplicitPath: kubeconfigPath},
+		&clientcmd.ConfigOverrides{}).ClientConfig()
+	if err != nil {
+		log.Fatal(err)
+	}
+	client, err := clientset.NewForConfig(&restclient.Config{
```

Comment:
isn't 'kubeConfig' already a restclient.Config? Can you not pass it directly.

Also, you could use `clientset.NewForConfigOrDie()` since we're treating this as fatal anyway.